widgetastic_manageiq/__init__.py

```diff
@@ -1089,6 +1091,125 @@ def workaround_save_issue(self):
             "{}._handlers.change.map(function(handler) {{ handler() }});".format(self.item_name))
 
 
+class SSUIVerticalNavigation(VerticalNavigation):
+    """The Patternfly Vertical navigation."""
+    CURRENTLY_SELECTED = './/li[contains(@class, "active")]/a/span'
+
+    @property
+    def currently_selected(self):
+        return [
+            self.browser.text(el)
+            for el
+            in self.browser.elements(self.CURRENTLY_SELECTED, parent=self)]
+
+
+class SSUIlist(Widget, ClickableMixin):
+    """Represents the list of items like Services , Orders or service catalogs in SSUI pages.
+    """
+
+    LOCATOR = '//div[@id={}]/../div//a[normalize-space(.)={}]'
+
+    def __init__(self, parent, list_name, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.list_name = list_name
+
+    def locator(self, item_name):
+        return self.LOCATOR.format(quote(self.list_name), quote(item_name))
+
+    def click_at(self, item_name):
+        """Clicks the list item with this name.
+
+        Args:
+            item_name: Name of the item
+        """
+        return self.browser.click(self.locator(item_name))
+
+
+class SSUIDropdown(Dropdown):
+    """Represents the SSUI dropdown.
+    Args:
+        text: Text of the button, can be the inner text or the title attribute.
+    """
+    ROOT = ParametrizedLocator(
+        '//button[contains(@class, "dropdown-toggle")]/span[normalize-space(.)="Configuration"]')
+    BUTTON_LOCATOR = '//button'
+    ITEMS_LOCATOR = './ul/li/a'
+    ITEM_LOCATOR = '//ul/li/a[normalize-space(.)={}]'
+
+    def __init__(self, parent, text, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.text = text
+
+
+class SSUIPrimarycard(Widget, ClickableMixin):
+    """
+        Represents a primary card on dashboard like Total Service or Total Requests.
+    """
+
+    LOCATOR = '//div[@class="ss-dashboard__card-primary__count"]/h3[normalize-space(.)={}]'
+    COUNT = '//div[@class="ss-dashboard__card-primary__count"]/h2'\
+            '[./following-sibling::h3[normalize-space(.)={}]]'
+
+    def __init__(self, parent, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+
+    def locator(self, item_name):
+        return self.LOCATOR.format(quote(item_name))
+
+    def count(self, item_name):
+        return self.COUNT.format(quote(item_name))
+
+    def click_at(self, item_name):
+        """Clicks the list item with this name.
+
+        Args:
+            item_name: Name of the item
+        """
+        return self.browser.click(self.locator(item_name))
+
+    def get_count(self, item_name):
+        """
+        Returns the number of items displayed on UI.
+
+        Args:
+            item_name: Name of the item
+        """
+        return self.browser.text(self.count(item_name))
+
+
+class SSUIAggregatecard(Widget, ClickableMixin):
+    """Represents an aggregate card like Current Services or Retired services.
+    """
```

Comment:
OPTIONAL: all docstrings (not just this one) should follow the same style:
```python
"""Single line docstring fits on single line."""
```
and for multi-line docstrings:
```python
"""Summary line fits on single line.

More detailed description, arguments, etc.
"""
```
Details:
https://www.python.org/dev/peps/pep-0257/#one-line-docstrings
https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings